### PR TITLE
Re-enable osv-scanner on pushes to main

### PR DESCRIPTION
**Pull request by sealesj**
The osv-scanner workflow was disabled on pushes to main while testing the PR patching workflow. This change re-enables the scanner to run on pushes to main (excluding forks).

Successful run of the current vulnerability workflow here: https://github.com/flutter/engine/actions/runs/7561640856/job/20590736333

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: .github/workflows/third_party_scan.yml [hunk 1]
```diff
@@ -2,6 +2,8 @@ name: Third party deps scan
 on:
   # Only the default branch is supported.
   branch_protection_rule:
+  push:
+    branches: [ main ]
```
Comment (godofredoc) [inline .github/workflows/third_party_scan.yml:6]: Would this be non-blocker?

Comment (sealesj) [reply .github/workflows/third_party_scan.yml:6]: Yes this is non blocking because of `fail-on-vuln: false`

Comment (sealesj) [reply .github/workflows/third_party_scan.yml:6]: Here's the documentation for that flag https://github.com/google/osv-scanner/blob/main/.github/workflows/osv-scanner-reusable-pr.yml#L41

**Comment by sealesj** [comment]
@godofredoc Does this look good to you?

**Review by zanderso — APPROVED** [review]

**Review by godofredoc — APPROVED** [review]

**Pull request merged**